cuda_core/tests/test_memory.py

```diff
@@ -512,35 +518,52 @@ def test_buffer_dlpack_failure_clean_up():
     assert after == before
 
 
-@pytest.mark.parametrize("use_device_object", [True, False])
-def test_device_memory_resource_initialization(use_device_object):
-    """Test that DeviceMemoryResource can be initialized successfully.
-
-    This test verifies that the DeviceMemoryResource initializes properly,
-    including the release threshold configuration for performance optimization.
-    """
+def test_modern_device_memory_resource_initialization(memory_resource_factory_with_device):
     device = Device()
+    MR, MRops, use_device_object = memory_resource_factory_with_device
 
-    if not device.properties.memory_pools_supported:
+    if MR is DeviceMemoryResource and not device.properties.memory_pools_supported:
```

Comment:
I find the repetition across tests and large numbers of conditions very difficult to read and would find it hard to maintain these tests in general.

I think the parameterized fixture `memory_resource_factory_with_device` as-is is not the right approach to making the tests readable.

How about splitting up the tests based on types, along with a fixtures that depend on a (maybe new?) `device` fixture that can be used to invoke `pytest.skip` based on properties? (`device` appears to be used in all of these tests)

In other words, instead of branching, write a test for each branch and move as much of the skipping logic as you can out of the test and into fixtures.

The shared asserts are already repeated, and given that's true, it's more readable to repeat them without branching.